app/logic/bonner.py

```diff
@@ -56,18 +56,23 @@ def getBonnerCohorts(limit=None, currentYear=date.today().year):
         Return a dictionary with years as keys and a list of bonner users as values. Returns empty lists for
         intermediate years, or the last 5 years if there are no older records.
     """
-    years = list(BonnerCohort.select(BonnerCohort, User).join(User).order_by(BonnerCohort.year).execute())
+    bonnorCohorts = list(BonnerCohort.select(BonnerCohort, User).join(User).order_by(BonnerCohort.year).execute())
 
-    defaultStart = currentYear-4
-    firstYear = years[0].year if len(years) and years[0].year < defaultStart else defaultStart
+    firstYear = currentYear-4
+    if bonnorCohorts:
```

Comment:
The refactor looks good, but I think you could combine these two `if bonnorCohorts:` conditionals into one.